cli/rt/99_main.js

```diff
@@ -283,20 +283,45 @@ delete Object.prototype.__proto__;
     Object.defineProperties(globalThis, mainRuntimeGlobalProperties);
     Object.setPrototypeOf(globalThis, Window.prototype);
     eventTarget.setEventTargetData(globalThis);
-    // Registers the handler for window.onload function.
-    globalThis.addEventListener("load", (e) => {
-      const { onload } = globalThis;
-      if (typeof onload === "function") {
-        onload(e);
-      }
-    });
-    // Registers the handler for window.onunload function.
-    globalThis.addEventListener("unload", (e) => {
-      const { onunload } = globalThis;
-      if (typeof onunload === "function") {
-        onunload(e);
+
+    const handlerSymbol = Symbol("eventHandlers");
+
+    function makeWrappedHandler(handler) {
+      function wrappedHandler(...args) {
+        if (typeof wrappedHandler.handler !== "function") {
+          return;
+        }
+        return wrappedHandler.handler.call(this, ...args);
       }
-    });
+      wrappedHandler.handler = handler;
+      return wrappedHandler;
+    }
+    // TODO(benjamingr) reuse when we can reuse code between web crates
+    function defineEventHandler(emitter, name) {
+      // HTML specification section 8.1.5.1
+      Object.defineProperty(emitter, `on${name}`, {
+        get() {
+          return this[handlerSymbol]?.get(name)?.handler ?? null;
+        },
+        set(value) {
+          if (!this[handlerSymbol]) {
+            this[handlerSymbol] = new Map();
+          }
+          let handlerWrapper = this[handlerSymbol]?.get(name);
+          if (handlerWrapper) {
+            handlerWrapper.handler = value;
+          } else {
+            handlerWrapper = makeWrappedHandler(value);
+            this.addEventListener(name, handlerWrapper);
+          }
+          this[handlerSymbol].set(name, handlerWrapper);
+        },
+        configurable: true,
+        enumerable: true,
+      });
+    }
```

Comment:
It seems this code can be obtained using `const { defineEventHandler } = window.__bootstrap.webUtil;`